### DISCUSSION TITLE
Unable to install - is this still active?

### DISCUSSION THREAD

**Discussion by orohde** [robske110/IDDataLogger#63] [General]
Hi,

today I tried to install this software in a ubuntu VM.
First I had to find out that the installation script is not working, I had to download it and manually change the php8.0 to php8.3.
After that, the installation worked, but I was not able to connect to the db:

`Loading Logger...
15:15:31.557886 [INFO] Starting ID DataLogger Version v0.0.6-InDev (3c6cd0d92e40ec9defd06dc7efc0056a26c1e576)...
Loading Main...
15:15:31.558163 [INFO] Reading config...
15:15:31.558215 [INFO] Connecting to db...
Loading DatabaseConnection...
Loading QueryCreationHelper...
15:15:31.568073 [INFO] Initializing db tables...
Loading ErrorUtils...
15:15:31.568651 [CRITICAL] PDOException: "SQLSTATE[42501]: Insufficient privilege: 7 ERROR: permission denied for schema public LINE 1: CREATE TABLE IF NOT EXISTS carStatus ( ^" (42501) in "/home/oliver/IDDataLogger/src/vwid/Main.php" at line 96
15:15:31.568693 [EMERGENCY] CRASHED WHILE STARTING; TRYING TO SHUTDOWN SAFELY
15:15:31.568700 [INFO] Shutting down...
15:15:31.568704 [CRITICAL] Forcibly shutting down while starting!`

once I changed the .env file and the config to point to my other postgresql it seems to work and I get the following warning:

`15:30:49.213539 [INFO] Done. Startup took 2.4271869659424s
15:30:49.213624 [INFO] Ready!
15:30:49.213654 [INFO] Fetching car status...
15:30:50.386576 [NOTICE] Could not find expected key 'hvacWithoutExternalPower' in carStatusData. Unexpected changes in older or newer car software can cause this!
15:30:50.386662 [INFO] Writing new data for timestamp 2024-11-24T13:00:33`

when calling the URL then, I get the following error:

`Error: required envvar DB_HOST not set!`

but it is set. Are you able to help me? By the way my car is an ID.7 Pro

Regards
Oliver

**Comment by robske110** [comment]
Hi Oliver, this is still working/active, however currently the installation script and instructions are broken on Debian 12 (and also the newest Ubuntu release).
For the first issue you faced, the fix is simple (it is a change in how default permissions work in postgresql). I‘ll try to commit the fix soon.
https://github.com/robske110/IDDataLogger/issues/60#issuecomment-2436294908 (German, but just use either of the commands)

The second one is also a permission issue (the web part is trying to read a .env file that resides in the users home folder, but I still need to figure out how I want to resolve this properly):
It can currently be fixed by either adjusting permissions (undoes the security improvement in Debian 12, but irrelevant if you have nothing else on the system) or by copying the .env manually into the web directory. Both options (in that order) are described here: https://github.com/robske110/IDDataLogger/issues/62#issuecomment-2496099293

The warning can be safely ignored, some fields are not available on certain cars/trim levels or newer/older car software versions (or even completely gone from the API).

**Reply by robske110** [reply]
Also, the docker option should still work out of the box, if you prefer to just have it work :)

**Reply by orohde** [reply]
Vielen Dank für die Info, ich werde es am Wochenende ausprobieren. Ich wünsche eine gute Zeit.
